client/gc.go

```diff
@@ -229,21 +161,28 @@ func (a *AllocGarbageCollector) Collect(allocID string) error {
 
 	ar := gcAlloc.allocRunner
 	a.logger.Printf("[INFO] client: garbage collecting allocation %q", ar.Alloc().ID)
-	ar.Destroy()
 
+	a.destroyAllocRunner(ar)
 	return nil
 }
 
 // CollectAll garbage collects all termianated allocations on a node
 func (a *AllocGarbageCollector) CollectAll() error {
```

Comment:
[OT] Huh, this method actually never returns an error because all destroying is done async.